packages/react/src/components/Dropdown/Dropdown.tsx

```diff
@@ -318,9 +325,36 @@ class Dropdown extends AutoControlledComponent<Extendable<DropdownProps>, Dropdo
   }
 
   a11yStatusTimeout: any
+  charKeysPressedTimeout: any
 
   componentWillUnmount() {
     clearTimeout(this.a11yStatusTimeout)
+    clearTimeout(this.charKeysPressedTimeout)
+  }
+
+  componentDidMount() {
+    this.props.items && this.setFilteredItemsAndItemStrings()
+  }
+
+  componentDidUpdate(prevProps: DropdownProps, prevState: DropdownState) {
+    const items = this.props.items
+    const prevItems = prevProps.items
+    const value = this.state.value as ShorthandCollection
+    const prevValue = prevState.value as ShorthandCollection
+    const itemsHaveBeenUpdated =
+      (items && items.length !== prevItems.length) ||
+      _.difference(items, prevItems).length > 0 ||
+      _.difference(prevItems, items).length > 0
+    const valueHasBeenAddedOrRemoved =
+      (this.props.multiple && value.length !== prevValue.length) ||
+      _.difference(value, prevValue).length > 0 ||
+      _.difference(prevValue, value).length > 0
+    const searchQueryHasBeenUpdated =
+      this.props.search && this.state.searchQuery !== prevState.searchQuery
+
+    if (itemsHaveBeenUpdated || valueHasBeenAddedOrRemoved || searchQueryHasBeenUpdated) {
+      this.setFilteredItemsAndItemStrings()
```

Comment:
This will produce an additional render 😭 Please consider to use `static getDerivedStateFromProps()`, in that case you don't need `itemsHaveBeenUpdated` and other complicated logic, just always filter them

Reply:
will look into it, thank you!

Reply:
refactored with `getDerivedStateFromProps` and it works, it's just that we will get the error from React about the deprecated `componentWillReceiveProps` from the controlled component implementation.

leave it like this and create a task to fix that ?

Reply:
actually we need to fix this, it fails the build.